Godeps/_workspace/src/k8s.io/kubernetes/pkg/securitycontextconstraints/capabilities/mustrunas.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package capabilities
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/fielderrors"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// defaultCapabilities implements the CapabilitiesSecurityContextConstraintsStrategy interface
+type defaultCapabilities struct {
+	defaultAddCapabilities   []api.Capability
+	requiredDropCapabilities []api.Capability
+	allowedCaps              []api.Capability
+}
+
+var _ CapabilitiesSecurityContextConstraintsStrategy = &defaultCapabilities{}
+
+// NewDefaultCapabilities creates a new defaultCapabilities strategy that will provide defaults and validation
+// based on the configured initial caps and allowed caps.
+func NewDefaultCapabilities(defaultAddCapabilities, requiredDropCapabilities, allowedCaps []api.Capability) (CapabilitiesSecurityContextConstraintsStrategy, error) {
+	return &defaultCapabilities{
+		defaultAddCapabilities:   defaultAddCapabilities,
+		requiredDropCapabilities: requiredDropCapabilities,
+		allowedCaps:              allowedCaps,
+	}, nil
+}
+
+// Generate creates the capabilities based on policy rules.  Generate will produce the following:
+// 1.  a capabilities.Add set containing all the required adds (unless the
+// 		container specifically is dropping the cap) and container requested adds
+// 2.  a capabilities.Drop set containing all the required drops and container requested drops
+func (s *defaultCapabilities) Generate(pod *api.Pod, container *api.Container) (*api.Capabilities, error) {
+	requiredAdd := makeCapSet(s.defaultAddCapabilities)
+	requiredDrop := makeCapSet(s.requiredDropCapabilities)
+	containerAdd := sets.NewString()
+	containerDrop := sets.NewString()
+
+	if scHasCapAdd(container.SecurityContext) {
```

Comment:
are we using those helper functions anywhere else? this would probably be easier to read as:
```
containerAdd := sets.NewString()
containerDrop := sets.NewString()
if container.SecurityContext != nil && container.SecurityContext.Capabilities != nil {
  containerAdd = makeCapSet(container.SecurityContext.Capabilities.Add)
  containerDrop = makeCapSet(container.SecurityContext.Capabilities.Drop)
}
```